azurelinuxagent/common/cgroups.py

```diff
@@ -796,3 +779,27 @@ def set_memory_limit(self, limit=None, unit='megabytes'):
             fileutil.write_file(memory_limit_file, '{0}\n'.format(value))
         else:
             raise CGroupsException("Memory hierarchy not available in this cgroup")
+
+
+class CGroupsLimits(object):
+    def __init__(self, cgroup_name, cpu_threshold=None, memory_threshold=None):
+        self.cpu_limit = cpu_threshold if cpu_threshold else self.get_default_cpu_limits(cgroup_name)
+        self.memory_limit = memory_threshold if memory_threshold else self.get_default_cpu_limits(cgroup_name)
```

Comment:
The signature of the constructor changes if you add a new limit. Combine that with my earlier comment in the CGroups constructor, and I have to conclude the design of this constructor isn't right.

Suggestion:
```
def __init__(self, cgroup_name, thresholds=none):
    self.cpu_limit = thresholds["cpu"] if thresholds and "cpu" in thresholds else self.get_default_cpu_limits(cgroup_name)
    self.memory_limit = thresholds["memory"] if thresholds and "memory" in thresholds else self.get_default_memory_limits(cgroup_name)
```

Reply:
Yes. Corrected this. The new constructor of `CGroupsLimits` now takes the dictionary and thus keeps it cleaner.